api/src/main/java/org/openmrs/Condition.java

```diff
@@ -246,7 +246,7 @@ public void setAdditionalDetail(String additionalDetail) {
 	 *         set
 	 */
 	public Date getOnsetDate() {
-		return onsetDate;
+		return onsetDate != null ? (Date) onsetDate.clone() : new Date();
```

Comment:
Aren't you changing the original logic of this method?

Reply:
That thing popped up in my mind too, but previous reviewers suggested me to check for null values and also can you check [this](https://github.com/openmrs/openmrs-core/blob/master/api/src/main/java/org/openmrs/api/impl/ConditionServiceImpl.java#L70) statement here, it has similar behaviour for null values, so instead of letting the user to check for null values I thought it would be better if we do that in the method itself. What are your thoughts on this?

Reply:
If you want to change the logic of this method, it would be better to use a different pull request and let this stick to its original intention. 😊

Reply:
Yeah, I think it would be better to keep the original intention of this PR :+1: